docs/tremor-script/index.md

```diff
@@ -167,6 +167,59 @@ Record literals in `tremor-script` are syntactically equivalent to JSON document
 }
 ```
 
+#### Binary
+
+Binaries are based on the [Erlang bit syntax](https://erlang.org/doc/programming_examples/bit_syntax.html).
+
+Binary grammar:
+
+> ![record grammar](grammar/diagram/Binary.png)
+
+Binaryfields grammar:
+
+> ![fields grammar](grammar/diagram/BinaryFields.png)
+
+Binaryfield grammar:
+
+> ![field grammar](grammar/diagram/BinaryField.png)
+
+
+Parts of each field are: `<value>:<size>/<type>` where both `size` and `type` are optional.
+
+The types currently supported are:
+
+* `binary` - this can handle both binaries and strings, `size` here refers to the number of bytes
+* `integer` - this can represent integers, `size` here means size in bits. In addition the type can be specified with `big` and `little` for indianness and `singed` and `unsigned` for singedness.
+
+```tremor
+# constructing a TCP package
+# using made up, non correct values
+
+let event = {
+  "src": {"port": 1234},
+  "dst": {"port": 2345},
+  "seq": event,
+  "ack": 4567,
+  "offset": 1,
+  "res": 2,
+  "flags": 3,
+  "win": 4,
+  "checksum": 5,
+  "urgent": 6,
+  "data": "snot badger!"
+};
+
+<<
+  event.src.port:16,  event.dst.port:16,
+  event.seq:32,
+  event.ack:32,
+  event.offset:4, event.res:4, event.flags:8, event.win:16,
+  event.checksum:16, event.urgent:16,
+  event.data/binary
+>>
+
+```
+
```

Comment:
It would be useful to mention the possible operations on binary values, like comparison (and how these behave).
E.g. how to concat two binary values, how to prefix something etc.

Reply:
You are right! It felt inconsistent to include it only for binary, so I extended the operator table to list the types for each operator